pgjdbc/src/main/java/org/postgresql/core/Encoding.java

```diff
@@ -148,7 +148,9 @@ public static Encoding getDatabaseEncoding(String databaseEncoding) {
     String[] candidates = encodings.get(databaseEncoding);
     if (candidates != null) {
       for (String candidate : candidates) {
-        LOGGER.log(Level.FINEST, "Search encoding candidate {0}", candidate);
+        if (LOGGER.isLoggable(Level.FINEST)) {
+          LOGGER.log(Level.FINEST, "Search encoding candidate {0}", candidate);
```

Comment:
This is `log(Level level, String msg, Object param1)`, and it does not require guards